lib/widgets/message_list.dart

```diff
@@ -345,6 +351,67 @@ class ScrollToBottomButton extends StatelessWidget {
   }
 }
 
+class MarkAsReadWidget extends StatelessWidget {
+  const MarkAsReadWidget({super.key, required this.model});
+
+  final MessageListView model;
+
+  void _handlePress(BuildContext context) async {
+    if (!context.mounted) return;
+    final narrow = model.narrow;
+    final connection = model.store.connection;
+    if (connection.zulipFeatureLevel! >= 155) {
```

Comment:
should be marked with a todo-server comment

For background and motivation on those comments, see here: https://github.com/zulip/zulip-mobile/blob/main/docs/style.md#todo-server
though the formatting has evolved slightly since then; see examples in this codebase.